EIPS/eip-5334.md

```diff
@@ -0,0 +1,113 @@
+    function userOf(uint256 tokenId) external view returns(address);
+
+    /// @notice Get the user expires of an NFT
+    /// @dev The zero value indicates that there is no user
+    /// @param tokenId The NFT to get the user expires for
+    /// @return The user expires for this NFT
+    function userExpires(uint256 tokenId) external view returns(uint256);
+
+    /// @notice Get the user level of an NFT
+    /// @dev The zero value indicates that there is no user
+    /// @param tokenId The NFT to get the user level for
+    /// @return The user level for this NFT
+    function userLevel(uint256 tokenId) external view returns(uint256);
+}
+```
+
+The `userOf(uint256 tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `userExpires(uint256 tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `userLevel(uint256 tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `setUser(uint256 tokenId, address user, uint64 expires)` function MAY be implemented as `public` or `external`.
+
+The `UpdateUser` event MUST be emitted when a user address is changed or the user expires is changed or the user level is changed.
+
+The `supportsInterface` method MUST return `true` when called with `0xad092b5c`.
+
+## Rationale
+
+This model is intended to facilitate easy implementation. Here are some of the problems that are solved by this standard:
+
+### Clear Rights Assignment
+
+With Dual “owner” and “user” roles, it becomes significantly easier to manage what lenders and borrowers can and cannot do with the NFT (in other words, their rights). Additionally, owners can control who the user is and it’s easy for other projects to assign their own rights to either the owners or the users.
+
+### Simple On-chain Time Management
+
+Once a rental period is over, the user role needs to be reset and the “user” has to lose access to the right to use the NFT. This is usually accomplished with a second on-chain transaction but that is gas inefficient and can lead to complications because it’s imprecise. With the `expires` function, there is no need for another transaction because the “user” is invalidated automatically after the duration is over.
+
+### Easy Third-Party Integration
+
+In the spirit of permission less interoperability, this standard makes it easier for third-party protocols to manage NFT usage rights without permission from the NFT issuer or the NFT application. Once a project has adopted the additional `user` role and `expires` and `level`, any other project can directly interact with these features and implement their own type of transaction. For example, a PFP NFT using this standard can be integrated into both a rental platform where users can rent the NFT for 30 days AND, at the same time, a mortgage platform where users can use the NFT while eventually buying ownership of the NFT with installment payments. This would all be done without needing the permission of the original PFP project.
```

Comment:
Again, I think this is more Motivation.

Things you might want to put in the Rationale section could be:

 - Why `setUser` is part of the specification instead of leaving it undefined.
 - Why `level` is a `uint8`.
 - Why `level` is not an optional extension.
 - Why you only support a single `user` instead of multiple.